spec/features/user_sorts_todos_spec.rb

```diff
@@ -0,0 +1,103 @@
+require "rails_helper"
+
+feature ".index" do
+    before do
+        sign_in
+
+        create_todo "Exercise"
+        create_todo "Buy milk"
+    end
+
+    context "when selecting" do
+        it "alphabetical sort" do
+            select_sort("Sort alphabetically")
+
+            expect("Buy milk").to appear_before("Exercise")
+        end
+
+        it "newest to oldest" do
+            select_sort("Sort newest to oldest")
+
+            expect("Buy milk").to appear_before("Exercise")
+        end
+
+        it "oldest to newest" do
+            select_sort("Sort oldest to newest")
+
+            expect("Exercise").to appear_before("Buy milk")
+        end
+
+        it "reverse alphabetical" do
+            select_sort("Sort alphabetically DESC")
+
+            expect("Exercise").to appear_before("Buy milk")
+        end
+
+        it "no sort" do
+            select_sort("No sort")
+
+            expect("Exercise").to appear_before("Buy milk")
+        end
+    end
+end
+
+feature "edge case" do
```

Comment:
You are testing the right things, however we should follow best practices so it's easier for other developers to understand how your code works.

We wouldn't call the spec itself as "edge case". We should describe the spec at a high level here and then describe it in detail in the nested blocks.

`context` blocks usually gives more information about the environment in the spec and then `it` block asserts it

An example for showing edge cases in a spec.

```
describe '#destroy' do

  context 'when resource is found' do
    it 'responds with 200' do
      expect_something
    end
    it 'shows the resource' do
      expect_something
    end
  end

  context 'when resource is not found' do
    it 'responds with 404'do
      expect_something
    end
  end

  context 'when resource is not owned' do
    it 'responds with 404' do
      expect_something
    end
  end
end
```

Here the 'when resource is not found' and the 'when resource is not owned' are examples of edge cases when testing the `#destroy` method.

The blocks are meant to read like english. If you see here the `it` blocks are kind of like a sentence and the spec is readable. A sentence could be formed like "When resource is not found it responds with 404 and expects response to equal 404". Every spec does not HAVE to read like english but it's nice to get it close to readable.